Readme.md

```diff
@@ -117,16 +117,327 @@ with keys corresponding to the options described in [CLI Arguments](#cliargs):
+```javascript
+{
+  errorProps: ['*']
+}
+```
+
+<a id="options-level-first"><a>
+#### `levelFirst` (Boolean)
+
+Default: `false`
+
+Swaps the placement of the `time` and `level` keys in each log line.
+
+> This setting is ignored if the `format` property is set
+
+##### true
+```bash
+INFO [1522431328992] (42 on foo): hello world
+```
+##### false
+```bash
+[1522431328992] INFO (42 on foo): hello world
+```
+
+<a id="options-message-key"><a>
+#### `messageKey` (String)
+
+Default: `msg`
+
+The key in your log entry that will be used when appending the message to the first line
+
+<a id="options-translate-time"><a>
+#### `translateTime` (Boolean | String)
+
+Default: `false`
+
+Whether or not to translate the time from an epoch timestamp to a formated time string
+
+##### true
+```bash
+INFO [2018-03-30 10:35:28.992 -0700] (42 on foo): hello world
+```
+> Uses [dateformat](https://www.npmjs.com/package/dateformat) string of `'yyyy-mm-dd HH:MM:ss.l o'`
+
+##### string
+Will apply a [dateformat](https://www.npmjs.com/package/dateformat) string to the log entry timestamp
+```javascript
+{
+  translateTime: 'yyyy-mm-dd HH:MM:ss'
+}
+```
+will result in:
+```bash
+INFO [2018-03-30 10:35:28] (42 on foo): hello world
+```
+
+##### false
+Renders the epoch time
+```bash
+INFO [1522431328992] (42 on foo): hello world
+```
+
+<a id="options-search"><a>
+#### `search` (String)
+A [jmespath](http://jmespath.org/) query string to filter your logs.
+
+<a id="options-advanced"><a>
+### Advanced Options
+
+<a id="options-ignore-keys"><a>
+#### `ignoreKeys` (Array<String>)
+
+Default: `['v']`
+
+You can specify the log entry keys that should be ignored from output.  By default, all keys will be output except for the internal 'v' key entry.
+
+<a id="options-format"><a>
+#### `format` (Array<Object>)
+Allows formatting of the first log line by specifying the placement of delimiters and keys.  This allows you to tap into the default formatting logic of pino-pretty so you can easily customize your log entries based on the properties that you include in your loggers.
+
+The pino logger will automatically attempt to bind the following keys to a log entry:
+| key      	| description |
+|----------	|-----------	|
+| level    	| The log level |
+| time     	| Epoch timestamp |
+| name     	| The process name if available |
+| pid      	| The process pid |
+| hostname 	| The hostname of server where the application is running |
+| v        	| The pino log entry version 	|
+
+You can also bind as many properties to a log message or a logger via various apis.  This formatter will allow you to specify how each of the available keys will be logged in the first line, as well as how delimiters should be displayed depending on what log keys are available.
+
+##### key
+By specifiying a key entry in your format object, you are telling pino-pretty to attempt to output the value of this key in the log entry if available.
+
+```javascript
+interface KeyToken {
```

Comment:
What is an `interface`?  It looks to be in the future reserved words list, but isn't a feature of JavaScript -- https://www.ecma-international.org/ecma-262/9.0/index.html#sec-future-reserved-words

Reply:
That was me just trying to explain what each of the available tokens can have.  I was referencing TypeScripts syntax for those familiar with it.  An interface is an age old OOP construct.

Reply:
I was playing dumb. The docs should be readable by people familiar with JavaScript and no other language since this is a JavaScript based project. I haven't had the time or fortitude to read through all of this to make a better suggestion other than "this isn't it."